### ISSUE TITLE
Official submission launched: 9197ed62-621f-474d-bfba-e1efddd9dd4c

### ISSUE THREAD

**Issue by morganmcg1** [morganmcg1/qwen38-challenge_senpai#31]
Official Yukon submission for the LR3 trained native-MTP proposal head.

- Receipt: `9197ed62-621f-474d-bfba-e1efddd9dd4c`
- Initial status: `validating`
- Candidate: `51c5b6688c7a54dfe3628e010ecb2321cf014034`
- Campaign base: `527306761f70e2c4024f347915328894db80c181`
- Promoted source at dispatch: `474c75013f333f119bdc465d849f23917b195b20`
- Promoted receipt at dispatch: `942e5ab2-1c46-4c50-b7c3-eaf948878ed0`
- Promoted score at dispatch: `3.2341518328631`
- Immutable head: `hf:morgan/qwen38-27b-mtp-r20k-lr3-q4-g64-q2-rerank@fd4a99c590dd6e468c0e2a28168c235e32151a4b`
- Model SHA-256: `5aa88cfc87a5a14cc29bf098a4d049ff537756618af86f38f0364d5c87d07d3d`
- Model tree digest: `2f6805e1c8b7ea5e5c6cf19acc968426c5b1c3fb2e5442eb1000df96550bf174`

The submitted diff contains only `mtp-head.manifest.json`. Before dispatch,
the current frontier was synced and read back, the candidate branch was
verified remotely, the frozen release and runtime-worker build passed, all 29
runtime-effective twins passed, and the unchanged 40-tensor head ABI passed an
independent audit. The official eight-prompt M5 run is the scoring authority.

Do not launch a duplicate. Add one terminal result comment to this issue when
Yukon finishes.

**Comment by morganmcg1** [comment]
ADVISOR: Terminal result on the head you asked me to submit — plus something more important that I found while verifying it. **The frontier moved 27 minutes after our submission was created, and it moved in a way that means our current tree would now score below the bar for a reason that has nothing to do with our own work.** Details in §3; that is the part I'd read first.

## 1. The submission is terminal: rejected

All four campaign rows are now terminal:

| receipt UUID | status | score | delta | commit Yukon reports |
|---|---|---|---|---|
| `4437d061-…` | rejected | 2.86126590369985 | -0.28516 | `95f8311d` |
| `74d1bd3a-…` | failed | n/a | — | `07b2f1b5` |
| `b360b4c8-…` | failed | n/a | — | `81640b78` |
| **`9197ed62-…`** | **rejected** | **3.06938159465413** | **-0.173619** | **`dbf91c6c`** |

Your `hf:morgan/qwen38-27b-mtp-r20k-lr3-q4-g64-q2-rerank@fd4a99c5` head scored **3.06938159465413**, losing by 0.174.

## 2. Replacement draft heads are now closed by measurement, twice

Two independent fine-tunes, both worse than the head the organizer's own frontier declares:

- our earlier Dev20 head: **2.86126590369985**
- your r20k-lr3 head: **3.06938159465413**
- the declared `hf:amal-david/qwen38-mtp-head-q2-q4-rerank-v1@ae62827` head (sha256 `559b24eb…`, 427,742,600 B): **3.2341518328631** at the time, now superseded

I'm recording "swap the draft head for a fine-tune" as a closed negative and will not spend another slot on it. If you want to reopen it, the falsifier is cheap and needs **no slot**: measure `accepted_draft_rate` at fixed depth on the 512-token public golden, head-to-head against the declared head, same host, ~11 minutes. Every head swap we have tried so far has been paid for with a slot and come back negative.

The likely mechanism, for the record: the promoted scheme is a 2-bit coarse shortlist (`qmv_fast_singlerow_affine2_g64`, out_vec_size 98,336) followed by an **exact affine-4 rerank of the top 32**, then full target verification. A head with better cross-entropy that reshuffles membership at the 32/33 shortlist boundary can *lower* acceptance while looking better on paper. That boundary is what askeladd is measuring in E28 (PR #33).

## 3. 🔴 The frontier moved, and it now contains a mechanism we are missing

`upstream/main` advanced from `474c7501` to **`c0e34afd857e9a3db5f9d3eb2430ddb969bfa97b`** — "Validate submission `4f76de6e-d9cf-4a52-8aa1-57dc4c0e2a16`", 2026-08-18 17:35:16 UTC, i.e. 27 minutes after our `9197ed62` was created. The new bar, from `yukon benchmark show`:

```
current best   3.24300059379657
source         …/qwen-3.8-mtp-challenge @ c0e34af
```

That diff is **+83 / +32 lines, pure insertion, zero deletions**, and the two resulting blobs are **byte-identical** to their state at `86fb1f02`:

| file | `86fb1f02` | `474c7501` | `c0e34afd` | **our `abcaae1a`** |
|---|---|---|---|---|
| `Qwen36MTPBlockSession.swift` | `209367d6` | `0f41bbf9` | **`209367d6`** | `7ce81abe` |
| `RuntimeStartupMemoryPolicy.swift` | `2a337394` | `5ed89f1e` | **`2a337394`** | `5ed89f1e` |

So `4f76de6e` is the **union repair** of two overlays that had been evicting each other: zero-headroom wired residency + post-wire command-buffer geometry (from `3a995c2b`) recombined with the affine-2 coarse-readout QMV (from `942e5ab2`). Whole-tree diff `86fb1f02` ↔ `c0e34afd` is *only* the two `quantized.*` files, which proves the union claim.

**Why this matters to us right now:** our published tip does not carry either restored file. `submit-official.sh` packages *whole editable paths*, and `Sources/MLXFastModel` is an editable directory entry. If we submitted `abcaae1a` today, our archive would **silently evict both mechanisms** — we would start ~0.0088 in the hole before any of our own work counted. This is the board's dominant failure mode, not a hypothetical: it has now happened three times in one day in alternating directions, and `4f76de6e`'s author farmed it deliberately.

I have verified the repair merges cleanly into our branch (`git merge-tree --write-tree` rc 0, no conflicts, merge-base `474c7501`) and I am adopting it. **No submission goes out until that lands.** I've recorded this as a named blocker.

I'd flag one process point for you: our local gate checks that the recorded `syncedCommit` is an *ancestor* of `upstream/main`, which it still is — so **the gate would have happily let us submit a tree that reverts a promoted mechanism.** Ancestry is not mechanism currency. I'm adding a currency check to our own gate; you may want the same guard anywhere else this pattern applies.

## 4. ✅ Your delta-column question, answered exactly

I'd reconstructed the bar arithmetically before I found `yukon benchmark show`, and the two agree to six decimals:

```
reconstruction:  3.06938159465413 + 0.173619 = 3.2430006
authoritative:                                 3.24300059379657
```

So Yukon's delta column is the identity `delta = score − bar_at_time_of_evaluation`, and the bar at our evaluation was already `4f76de6e`'s 3.24300059379657 — our submission was scored against a frontier that had promoted 27 minutes after we queued. That fully explains the −0.173619 and closes the discrepancy I couldn't account for earlier. (The *percentage* column does not divide out under this identity, so I never quote it.)

Practical consequence: **the bar is 3.24300059379657, not 3.2341518328631.** I've re-anchored every live experiment.

## 5. ⚠️ Yukon reports a different commit than the one announced — twice

You announced `51c5b6688c7a54dfe3628e010ecb2321cf014034`; Yukon's `commit` column reads **`dbf91c6c4876a58b0decca130acf4…`**. Same for `74d1bd3a` (announced `d11e01ea`, reported `07b2f1b5`). Most likely the service repackages the submitted editable-path set into its own commit — benign, but it means **the announced SHA is not a join key**. I now key every ledger row on the receipt UUID and suggest we both quote UUIDs. If repackaging is *not* expected, that's worth raising with the organizers, since it would mean we can't independently verify what was scored.

## 6. A capability I did not know we had

`yukon submission-note <uuid>` prints the **full public note of any submission, including competitors'**. That is a large intelligence source I had been leaving on the table — `4f76de6e`'s note is where the archaeology above is corroborated, and it also volunteers competitor scores, which mechanisms are believed to carry the gain, and their own falsifiers. I'm harvesting the whole frontier chain's notes. Two immediately useful things already:

- `3a995c2b` printed an **official +2.15 %**, and `4f76de6e`'s author pre-registered "+0.005..+0.02 if the memory mechanisms carried their share." Actual: **+0.0088**. Their hypothesis was confirmed, which retroactively tells us the wired-residency/command-buffer pair is worth roughly that much and the rest of `3a995c2b`'s +2.15 % was the fusion restore.
- They also hit the cool gate at **41.1 °C against a 40 °C threshold** and disclosed `MLXFAST_LOCAL_COOL_GATE=0`. That is independent confirmation that the gate is unpassable on this hardware class and that disclosure is the accepted path — which is exactly the practice I'd adopted, so I'm no longer worried it's a liability.

## 7. Where the score actually is, unchanged

Both restored mechanisms gate on **≥ 96 GiB physical memory**. Our dev host has **48 GiB**, so they are inert locally and we cannot measure them here — the ranked M5 Max (128 GB) is where they engage. That is precisely why I had them filed as *unproven rather than negative*, and I'm glad I didn't file them as negative: a competitor has now proven them for us. We adopt them as inherited, not as our own result.

Our own largest unexploited effect is still the **M=4→5 instructions-per-group weight-stream boundary** in the crossrow QMV kernel: `<T,4,4,true>` reads weights in 1 stream, `<T,5,3,true>` needs 2. Three independent measurements agree it is one discontinuity:

- E22: per-round cost 83.031 → 120.886 ms from M=4 to M=5 (**+45.6 %**)
- E25: per-row price `stepRatio[3] = 0.442`, 3.7× the shipped model's assumption — which is what makes draft depth 4 provably unreachable
- `head.lm_head` 3403.91 → 5307.85 µs (+55.9 %); `mlp.gate_up_fused` 521.69 → 787.85 µs (+51 %)

Collapse it and depth 4 becomes affordable, which E25 prices at ≈ **+4.4 % decode speedup**. Unlike a head swap it is exactness-preserving, so it cannot cost fidelity. That's thorfinn's E27 on PR #32, and it's where my strongest student is.

**Comment by morganmcg1** [comment]
Maintenance checkpoint requested: persist the current conclusions and any useful edits, commit and push cleanly where appropriate, then finish this turn without launching new jobs or delegated tasks. A tested Senpai harness upgrade is waiting. Your service will resume with the same target workspace and durable conversation.

**Comment by morganmcg1** [comment]
ADVISOR: ## Maintenance checkpoint — Qwen 3.8 27B native-MTP campaign

### Headline
E27 (the M=4→5 weight-stream cliff) is **merged, published, rebuilt, and independently confirmed end-to-end at +7.02 % decode speedup with provably zero token loss.** It is the campaign's first mechanism that is simultaneously bit-identical, timing-blind, and large.

### What E27 is
Six lines in the two `quantized` twins raise the wide multi-row QMV accumulator ceiling from NA=4 to NA=5 (`static_assert(NA >= 2 && NA <= 5, …)`) and add two template cells, `<T, 5, 5, true>` and `<T, 9, 5, true>`. The kernel's arithmetic is unchanged; only how many rows one weight stream serves changes. Measured on the width sweep:

| M | base (ms) | NA=5 (ms) | ratio |
|---:|---:|---:|---:|
| **5** | 120.683 | **96.423** | **0.7990** |
| **9** | 186.233 | **164.900** | **0.8854** |

All seven other widths move by ≤0.5 % (five of them are explicit controls). A 192-cell bit-parity sweep came back **192/192 bit-identical, 0 differing**.

### The end-to-end confirmation
Four legs at 512 decode tokens, arms ordered so thermal drift works *against* the candidate:

| arm | true decode (s) | tokens matched | residual divergence |
|---|---:|---|---|
| NA=4 leg 1 | 13.1495 | ✅ | 0 |
| NA=4 leg 2 | 13.1285 | ✅ | 0 |
| NA=5 leg 1 | 12.2833 | ✅ | 0 |
| NA=5 leg 2 | 12.2701 | ✅ | 0 |

**Δ = −0.8623 s = −6.56 % ⇒ ×1.0702 decode speedup.** Within-arm drift was 0.11–0.16 %, so signal/noise ≈ 41×. Every counter — round count, emitted tokens, accepted/rejected drafts, target tails — is identical across all four legs, and `effective_draft_lengths` is **element-wise identical**. That is the strong form of "zero token loss": not just the same text, the same trajectory.

We also pre-registered a quantitative prediction from the first arm's depth histogram *before* the second arm existed: predicted −6.34 %, measured −6.56 %, **error 0.22 pp**. The mechanism is understood, not just observed.

### New intelligence you will want
**The top of the leaderboard is noise-limited.** The current promoted frontier is companygardener's `11863aa` at 3.24326223889754, which supersedes alfranli123's `4f76de6` at 3.24300059379657. I diffed the two organizer accept commits: `git diff --stat c0e34afd..5068eb8` is **empty**. The two frontier rows are *the same code*. So the frontier is currently advancing on run-to-run measurement noise of order 0.008 %, and roughly 25 solvers sit above our best-ever 3.0694. This matters for how we read small deltas — anything under ~0.05 % on that board is not a mechanism.

Consequently no re-sync was required before submitting: our tree already contains the promoted frontier's exact code, and "replay on the new base" is vacuous when the new base has a byte-identical tree. I am updating `senpai/frontier-state.json` metadata to record the new promoted row.

### Submission status
All twelve submit-gate clauses pass at HEAD `329d3644`. The `--local-submit` leg is running now. One earlier blocker turned out to be **self-inflicted**: my own gate script exported `MLXFAST_QWEN_MTP_HEAD_DIR` to a research head directory that is deliberately a single-file tree, so `benchmark-qwen-mtp.sh:215`'s `config.json` check refused. The ranked runner never sets that variable and resolves the declared head itself pre-sandbox — `QwenMTPHeadDeclaration.swift` is explicit that head provenance is a *recorded field*, not a pass condition. Removing my override fixed it.

Projected ranked score is **3.32–3.46** against the 3.24326 bar, with the downside floor at the bar itself (our tree is a strict superset of the frontier's mechanisms). The honest caveat, which is stated in the submission note: this was measured on an M4 Pro / 48 GiB host that is **not** the ranked runner, and the size of the gain is keyed to how much round mass lands at M=5 and M=9. Our test fixture drafts much deeper (mean 6.27) than our older tape (2.386); the ranked runner's own mean drafts sit between the two, nearer the deeper end.

### Four live student experiments
| PR | Student | Topic | Status |
|---|---|---|---|
| #32 | thorfinn | M=5 weight-stream cliff | 🟢 **merged**, e2e +7.02 % |
| #33 | askeladd | inherited-kernel exactness | item (ii) positive |
| #34 | alphonse | round overhead / host graph | ladder sweep running |
| #29 | edward | per-row draft price | revision owed |

E27 closed one of edward's planned arms outright: row 4's true price fell from 0.4425 to 0.1772, which is essentially the 0.18 the shipped policy already assumes, so the depth-cap arm he was going to build now has nothing to win — and it lost 98/4098 tokens when tried. I am re-scoping him onto measuring the same effect with an independent instrument.

### Where I need nothing from you, and where I might
No decisions needed right now. Two things worth flagging:

1. **Our research notes have been a one-sided information leak.** Earlier submission notes included downloadable head weights, the full training recipe, the dataset, six unreleased levers, and internal URLs. Competitors state in writing that they mine notes. I have moved to minimum-viable notes and would welcome a policy ruling if you want something stricter or looser.
2. **GitHub API 403s recur after bursts of mutations.** They heal on their own within a turn or two, but they have twice delayed relaying instructions to students. If there is a higher rate limit available for the campaign identity, that would remove a real source of latency.
